report/src/main/java/com/hartwig/actin/report/pdf/tables/molecular/MolecularDriversGenerator.java

```diff
@@ -69,7 +69,7 @@ public Table contents() {
             table.addCell(Cells.createContent(nullToEmpty(entry.bestResistanceEvidence())));
         }
 
-        if (hasPotentiallySubclonalVariants(molecular.drivers().variants())) {
+        if (!entries.isEmpty() && hasPotentiallySubclonalVariants(molecular.drivers().variants())) {
```

Comment:
I don't quite like this, entries and "molecular driver variants" are different datamodels. If you go into the factory creating the driver entries you can see that only "reported" variants get converted. This means that previously, unreported variants would lead to a footer note without actually being present. Also, your fix doesn't fix every situation since there could be other entries (like amps and dels) that would never be subclonal.

Maybe a better solution is to add an explicit "isPotentiallySubclonal" to MolecularDriverEntry datamodel, and add the asterix within the table rather than the datamodel factory, and subsequently the decision to add footnote yes/no becomes trial as well?

Reply:
Good point about the other entries that could produce the footnote as well. We should be able to unambiguously determine if the footnote is needed by filtering the set of variants, i.e.:
```
        boolean hasPotentiallySubClonalVariants = molecular.drivers().variants().stream()
                .filter(Driver::isReportable)
                .anyMatch(ClonalityInterpreter::isPotentiallySubclonal);
```

I'll open a new PR to address this concern.

Reply:
Thanks!